### PR TITLE
add request and response with map type

### PR DESCRIPTION
**Pull request by omarxp**
- add request and response with map type
- update midtrans-new-3ds

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -166,10 +166,10 @@ type ConvStoreDetail struct {
 	Message string `json:"message"`
 }
 
-// GopayDetail: Represent gopay detail
+// GopayDetail : Represent gopay detail
 type GopayDetail struct {
 	EnableCallback bool   `json:"enable_callback"`
-	CallbackUrl    string `json:"callback_url"`
+	CallbackURL    string `json:"callback_url"`
```
Comment (rizdaprasetya) [inline request.go]: Let's keep `CallbackUrl` to avoid backward compatibility issue to merchant who already use it.

File: snap.go [hunk 2]
```diff
@@ -50,3 +50,27 @@ func (gateway *SnapGateway) GetToken(r *SnapReq) (SnapResponse, error) {
 
 	return resp, nil
 }
+
+// GetTokenQuickMap : Quickly get token without constructing the body manually
+func (gateway *SnapGateway) GetTokenQuickMap(orderID string, grossAmount int64) (ResponseWithMap, error) {
```
Comment (rizdaprasetya) [inline snap.go]: Let's use `GetTokenQuickWithMap` as name

File: snap.go [hunk 3]
```diff
@@ -50,3 +50,27 @@ func (gateway *SnapGateway) GetToken(r *SnapReq) (SnapResponse, error) {
...
+	return gateway.GetTokenMap(&SnapReqWithMap{
+		"transaction_details": TransactionDetails{
+			OrderID:  orderID,
+			GrossAmt: grossAmount,
+		},
+	})
+}
+
+// GetTokenMap : Get token by consuming SnapReqWithMap
+func (gateway *SnapGateway) GetTokenMap(r *SnapReqWithMap) (ResponseWithMap, error) {
```
Comment (rizdaprasetya) [inline snap.go]: Let's use `GetTokenWithMap` as name

**Comment by rizdaprasetya** [comment]
Also example usage of these new function should be added as example on `Readme.md` file

**Comment by omarxp** [comment]
Hi @rizdaprasetya, already updated.

**Pull request merged**